src/goto-programs/goto_program.h

```diff
@@ -539,8 +539,11 @@ class goto_programt
       return nullopt;
     }
 
-    return std::next(
+    auto location_target = std::next(
       instructions.begin(), location_number - start_location_number);
+
+    CHECK_RETURN(location_target->location_number == location_number);
```

Comment:
I think this was one of the suggestions by Michael. I think he meant this should be done before `std::next` happens, as it can crash without the warrantied property this is assuming (that locations are contiguous).

Reply:
https://github.com/diffblue/cbmc/pull/4150/files#r255405529 .. I think `std::next` would only crash if we wanted it to move the iterator beyond `end()`.